skimage/transform/tests/test_distance_transform.py

```diff
@@ -0,0 +1,123 @@
+
+
+def test_1d():
+    case = [1, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0]
+    out_euc = [1.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0]
+    out_man = [1.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0]
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow"),np.asarray(out_euc))
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow", dist_func=manhattan_dist, dist_meet=manhattan_meet),np.asarray(out_man))
+
+
+def test_2d():
+    case = [[1, 1, 1, 1, 1],
+            [1, 1, 0, 0, 0],
+            [1, 1, 1, 0, 0],
+            [1, 0, 1, 0, 1],
+            [0, 0, 1, 1, 1]]
+
+    out_euc = [[5.0, 2.0, 1.0, 1.0, 1.0],
+               [4.0, 1.0, 0.0, 0.0, 0.0],
+               [2.0, 1.0, 1.0, 0.0, 0.0],
+               [1.0, 0.0, 1.0, 0.0, 1.0],
+               [0.0, 0.0, 1.0, 1.0, 2.0]]
+
+    out_man = [[3.0, 2.0, 1.0, 1.0, 1.0],
+               [2.0, 1.0, 0.0, 0.0, 0.0],
+               [2.0, 1.0, 1.0, 0.0, 0.0],
+               [1.0, 0.0, 1.0, 0.0, 1.0],
+               [0.0, 0.0, 1.0, 1.0, 2.0]]
+
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow"),np.asarray(out_euc))
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow", dist_func=manhattan_dist, dist_meet=manhattan_meet),np.asarray(out_man))
+
+
+def test_3d():
+    case = [[[1, 0, 1],
+             [1, 1, 1],
+             [0, 0, 1]],
+            [[1, 1, 0],
+             [0, 1, 1],
+             [0, 1, 1]],
+            [[1, 1, 0],
+             [0, 1, 1],
+             [0, 1, 1]]]
+
+    out_euc = [[[1.0, 0.0, 1.0],
+                [1.0, 1.0, 2.0],
+                [0.0, 0.0, 1.0]],
+               [[1.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0],
+                [0.0, 1.0, 2.0]],
+               [[1.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0],
+                [0.0, 1.0, 4.0]]]
+
+    out_man = [[[1.0, 0.0, 1.0],
+                [1.0, 1.0, 2.0],
+                [0.0, 0.0, 1.0]],
+               [[1.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0],
+                [0.0, 1.0, 2.0]],
+               [[1.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0],
+                [0.0, 1.0, 2.0]]]
+
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow"),np.asarray(out_euc))
+    np.testing.assert_allclose(generalized_distance_transform(np.asarray(case),func="slow", dist_func=manhattan_dist, dist_meet=manhattan_meet),np.asarray(out_man))
+
+def test_euclidean_equivalent_to_ndimage():
+    from scipy.ndimage.morphology import distance_transform_edt, distance_transform_cdt
+    import time
+    case = (1+-1*(np.random.randint(50, size=(512,512,512))//48)).astype('float64')
+
+    start = time.time()
+    out_euc = distance_transform_edt(case)
+    print('scipy time:', time.time()-start)
```

Comment:
don't print in tests.

If you want to compare timings, do so in the benchmarks.